test/tests/functional/pbs_qsub_opts_args.py

```diff
@@ -86,6 +86,54 @@ def validate_error(self, err):
                 "ERROR in submitting a job with future time: %s" %
                 err.msg[0])
 
+    def jobdir_shared_body(self, location):
+        """
+        Test submission of job with sandbox=PRIVATE,
+        and moms have $jobdir_root set to shared.
+        """
+        self.remove_jobdir = True
+        momA = self.moms.values()[0]
+        momB = self.moms.values()[1]
+
+        loglevel = {'$logevent': 4095}
+        momB.add_config(loglevel)
+        c = {'$jobdir_root': '%s shared' % location}
+        for mom in [momA, momB]:
+            mom.add_config(c)
+            mom.restart()
+
+        a = {'Resource_List.select': '2:ncpus=1',
+             'Resource_List.place': 'scatter',
+             ATTR_sandbox: 'PRIVATE',
+             }
+        j = Job(TEST_USER, attrs=a)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
```

Comment:
On slow machines, 10s might not be enough.  You might start checking for the job to be in R after it is finished.

Reply:
@bhroam : I can increase the sleep_time to 60 seconds.

Reply:
I'd go with 30-45s.  60 is probably a little too long.

Reply:
@bhroam: ok, I'll update to 30.

Reply:
An alternative is to leave the sleep time default, and when you want the job to go away, just delete it.  This will mean it will run fast on fast machines, but also not fail on slow machines.  It changes the test though.  It's up to you.

Reply:
@bhroam : I'd rather just remove the set_sleep_time(10) and just use the default sleep (I think it's 30 seconds) and not delete the job, since I want the job to normally execute all the way to the end, and not be interrupted by a qdel.

Reply:
The default sleep is 100s.  If you want to not delete the job, set the sleep time to 30s.

Reply:
Ok, 30 seconds then.